js/data/custom_store.d.ts

```diff
@@ -38,10 +44,18 @@ export interface CustomStoreOptions<
     /**
      * @docid
      * @type_function_param1 options:LoadOptions
-     * @type_function_return Promise<any>|Array<any>
+     * @type_function_return Promise<Array|object>|Array
      * @public
      */
-    load: ((options: LoadOptions<TItem>) => PromiseLike<TItem> | Array<TItem>);
+    load: ((options: LoadOptions<TItem>) =>
+      | DxPromise<Array<TItem>
+          | {
+              data: Array<TItem> | Array<GroupItem>;
+              totalCount?: number;
+              summary?: Array<number>;
```

Comment:
number should replaced to unknown or any because summary values can have different types.